security/http/codegen.go

```diff
@@ -52,163 +73,221 @@ type SchemeData struct {
+// SecureRequestDecoders initializes the security attributes for HTTP request decoders.
+func SecureRequestDecoders(r *httpdesign.RootExpr, f *codegen.File) {
+	needsImport := false
+	for _, s := range f.Section("server-handler-init") {
+		needsImport = true
+		data := s.Data.(*httpcodegen.EndpointData)
+		e := r.Service(data.ServiceName).Endpoint(data.Method.Name)
+		if len(design.Requirements(e.Service.Name(), e.Name())) > 0 {
+			s.Source = strings.Replace(s.Source, "{{ .RequestDecoder }}", "Secure{{ .RequestDecoder }}", -1)
+		}
+	}
+	if needsImport {
+		codegen.AddImport(f.SectionTemplates[0],
+			&codegen.ImportSpec{Path: "goa.design/plugins/security"})
+	}
+	var schemes []*SchemeData
+	for _, s := range f.Section("request-decoder") {
+		data := s.Data.(*httpcodegen.EndpointData)
+		e := r.Service(data.ServiceName).Endpoint(data.Method.Name)
+		schemes = getSchemesForEndpoint(e)
+		if len(schemes) == 0 {
+			continue
+		}
+		f.SectionTemplates = append(f.SectionTemplates, &codegen.SectionTemplate{
+			Name:   "secure-request-decoder",
+			Source: authDecoderT,
+			Data: &SecureDecoderData{
 				SecureRequestDecoder: "Secure" + data.RequestDecoder,
 				RequestDecoder:       data.RequestDecoder,
 				PayloadType:          data.Payload.Ref,
 				Schemes:              schemes,
+				ServiceName:          data.ServiceName,
+				MethodName:           data.Method.Name,
+			},
+			FuncMap: codegen.TemplateFuncs(),
+		})
+	}
+}
+
+// SecureRequestEncoders initializes the security attributes for HTTP encoders.
+func SecureRequestEncoders(r *httpdesign.RootExpr, f *codegen.File) {
+	needsImport := false
+	for _, s := range f.Section("client-endpoint-init") {
+		needsImport = true
+		data := s.Data.(*httpcodegen.EndpointData)
+		e := r.Service(data.ServiceName).Endpoint(data.Method.Name)
+		if len(design.Requirements(e.Service.Name(), e.Name())) > 0 {
+			s.Source = strings.Replace(s.Source, "{{ .RequestEncoder }}", "Secure{{ .RequestEncoder }}", -1)
+		}
+	}
+	if needsImport {
+		codegen.AddImport(f.SectionTemplates[0],
+			&codegen.ImportSpec{Path: "goa.design/plugins/security"})
+	}
+	var schemes []*SchemeData
+	for _, s := range f.Section("request-encoder") {
+		data := s.Data.(*httpcodegen.EndpointData)
+		e := r.Service(data.ServiceName).Endpoint(data.Method.Name)
+		schemes = getSchemesForEndpoint(e)
+		if len(schemes) == 0 {
+			continue
+		}
+		f.SectionTemplates = append(f.SectionTemplates, &codegen.SectionTemplate{
+			Name:   "secure-request-encoder",
+			Source: authEncoderT,
+			Data: &SecureEncoderData{
+				SecureRequestEncoder: "Secure" + data.RequestEncoder,
+				RequestEncoder:       data.RequestEncoder,
+				PayloadType:          data.Payload.Ref,
+				Schemes:              schemes,
+				ServiceName:          data.ServiceName,
+				MethodName:           data.Method.Name,
+			},
+			FuncMap: codegen.TemplateFuncs(),
+		})
+	}
+}
+
+// getSchemesForEndpoint collects the security schemes for the given endpoint.
```

Comment:
style nit-pick: Go tends to favor concise names especially for private methods. How about `computeSchemes` instead of `getSchemesForEndpoint`?